quixstreams/kafka/producer.py

```diff
@@ -151,11 +154,16 @@ def flush(self, timeout: Optional[float] = None) -> int:
         Wait for all messages in the Producer queue to be delivered.
 
         :param float timeout: time to attempt flushing (seconds).
-            None or -1 is infinite. Default: None
+            None use producer default or -1 is infinite. Default: None
 
         :return: number of messages remaining to flush
         """
-        return self._producer.flush(timeout=timeout if timeout is not None else -1)
+        if timeout is None:
+            timeout = self._flush_timeout
+        if timeout is None:
+            timeout = -1
```

Comment:
It looks like one of the `None` checks can be moved to the `Producer.__init__` to be done once.